sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignatureCredential.cs

```diff
@@ -84,5 +84,22 @@ public SharedAccessSignatureCredential(SharedAccessSignature signature)
         ///
         public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext,
                                                              CancellationToken cancellationToken) => new ValueTask<AccessToken>(GetToken(requestContext, cancellationToken));
+
+        internal void UpdateEventHubResourceAndValidity(string eventHubResource, TimeSpan? signatureValidityDuration)
```

Comment:
I'm not sure that I understand the scenario where this would be used.   We definitely don't want to encourage allowing credentials to roam across Event Hub instances.